front/src/helpers/index.ts

```diff
@@ -0,0 +1,75 @@
+import { Todo, TodoDTO } from '../types'
+
+const parseTodoDTO = (data: {[propName: string]: any}) => {
+  const todoDTO = {} as TodoDTO;
+
+  // check id
+  if('id' in data) {
```

Comment:
looks interesting :) 

Yeah, that's the point - we can add additional logic while transforming server response into our model, just normally we don't try to guess the field. Code is ok, I just cannot imagine why our backend might return data in different shape

Reply:
@skyboyer I just tried to imagine what exact additional logic it would be. Because if we don't have so then we don't need `parseTodoDTO()` at all. We just can pass the response to `fromDTO()` immediately.

Reply:
well, `if('id' in data)` + other checks for fields you know should be there would be reasonable enough. I mean not several checks to field `id` field, but check for `id`, check for `title` so on.

With `zod` it could be 

```
const todoDTO = zod.object({
  id: zod.string(),
  title: zod.string(),
  isDone: zod.boolean()
}).parse(data);
```

Reply:
@skyboyer Do I need to use `zod` then? Or I shoud add simple type checking manually?
```
  if(('id' in data) && (typeof data.id === 'string')) {
    todoDTO.id = data.id;
  } else {
    throw new Error('There is no todo id or its type is not a string')
  }
```

Reply:
I propose to keep moving with checking manually and revisit this/play with zod(or any alternative io-ts, runtypes, probably yup - not sure if they are complete competitors) after everything else is done